application-diagram-api/src/main/java/com/xwiki/diagram/internal/handlers/DiagramLinkHandler.java

```diff
@@ -196,6 +199,19 @@ public String getMxCellNodeLink(String value)
         return null;
     }
 
+    /**
+     * If in a link cell there is also simple text, that will also be added in mxCell value, before or after the
+     * <a></a> tag. This needs to be cleaned, since just the link tag is needed.
+     *
+     * @param value value attribute of mxCell node
+     * @return just the link element inside
+     */
+    public String cleanLinkValue(String value)
+    {
+        String valueInsideLinkElement = StringUtils.substringBetween(value, "<a", "</a>");
+        return String.format("<a %s </a>", valueInsideLinkElement);
+    }
```

Comment:
This feels a bit hackish. It's not clear to me why you need this since you parse the HTML in ``getLinkFromEmbeddedNode()`` anyway. So it shouldn't matter if there is text before or after the link. Couldn't you just replace:

    ((Element) doc.getFirstChild()).getAttribute(HREF);

with a more bullet-proof way of getting the anchor element (e.g. using XPath and finding the first anchor element).

Reply:
For an element like https://github.com/xwikisas/application-diagram/pull/146/files#diff-a28d191cad4efd2ff1d7c794c89e41ddR733 were there are text nodes outside html elements
the value that will get here is `Simple text <br> <a href="http://localhost:8080/xwiki/bin/view/ATest/">Text</a>` and this is not well formed xml, so it will throw an error  in getLinkFromEmbeddedNode.
Thus this method have the purpose to remove any outside element.

Reply:
I think you just need to wrap ``value`` in ``<div>`` and ``</div>`` to make sure you have a well-formed XML no? And if the problem is the ``<br>`` then it would be safer to use the [``HTMLCleaner``](https://github.com/xwiki/xwiki-commons/blob/master/xwiki-commons-core/xwiki-commons-xml/src/main/java/org/xwiki/xml/html/HTMLCleaner.java) to parse the HTML.

Reply:
I added a fix using HTMLCleaner in my last commit. The problem with it is that there could be multiple `a` nodes inside the `value` that we get in java, but I would prefer to work on a different issue for that, since it doesn't seems that much related to the title of this issue anymore.

Reply:
OK, sounds good.